File: example/ios/SimpleCryptoExample/ResourcePath.m
//
//  ResourcePath.m
//  SimpleCryptoExample
//
//  Created by Diego Mello on 17/06/25.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ResourcePath, NSObject)
RCT_EXTERN_METHOD(getResourcePath:(NSString *)resourceName
                  ext:(NSString *)ext
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
@end

File: example/ios/ResourcePath.swift
import Foundation
import React

@objc(ResourcePath)
class ResourcePath: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }

  @objc
  func getResourcePath(_ resourceName: String, ext: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    if let path = Bundle.main.path(forResource: resourceName, ofType: ext) {
      resolve(path)
    } else {
      reject("not_found", "Resource not found", nil)
    }
  }
} 